src/cargo/ops/vendor.rs

```diff
@@ -282,21 +339,24 @@ fn sync(
                 branch,
                 tag,
                 rev,
-                replace_with: merged_source_name.to_string(),
+                replace_with,
             }
         } else {
             panic!("Invalid source ID: {}", source_id)
         };
-        config.insert(name, source);
+
+        config.insert(name.to_string(), source);
     }
 
     if !config.is_empty() {
-        config.insert(
-            merged_source_name.to_string(),
-            VendorSource::Directory {
-                directory: opts.destination.to_path_buf(),
-            },
-        );
+        if !opts.no_merge_sources {
```

Comment:
It looks like this logic is otherwise now handled on each iteration of the loop for `no_merge_sources`, so could this logic also move into the loop? (I think that could end up removing this here)